src/charm.py

```diff
@@ -56,6 +68,31 @@ def apply_crds(self):
         self.unit.status = MaintenanceStatus("Applying CRDs")
         self.kubectl("apply", "-f", "templates/crd.yaml")
 
+    def apply_grafana_agent(self, remote_url):
+        patch_res = [
+            {"kind": "deployment", "name": "kube-ovn-monitor", "port": '"10661"'},
+            {"kind": "daemonset", "name": "kube-ovn-pinger", "port": '"8080"'},
+            {"kind": "daemonset", "name": "kube-ovn-cni", "port": '"10665"'},
+        ]
+        if not self.stored.prometheus_patched:
+            self.patch_prometheus_resources(patch_res)
+            self.stored.prometheus_patched = True
+
+        agent_config = self.render_template(
+            "grafana_configmap.yaml.j2",
+            model_name=self.model,
+            model_uuid=self.model.uuid,
+            application=self.model.app.name,
+            remote_url=remote_url,
+        )
+        self.kubectl("apply", "-n", "kube-system", "-f", agent_config)
+
+        if self.stored.grafana_agent_configured:
+            self.kubectl("delete", "-f", "templates/grafana_agent.yaml")
+        self.kubectl("apply", "-f", "templates/grafana_agent.yaml")
```

Comment:
So nothing needs to be rendered in `grafana_agent.yaml`, right?

Reply:
Yes, the only templates are the Grafana Agent configMap (`grafana_configmap.yaml.j2`) and the Prometheus patches (`patch-prometheus.yaml.j2`) for Kube-OVN resources. The `grafana_agent.yaml` manifest contains the resources needed to run the agent, the configuration is rendered (using the `grafana_configmap.yaml.j2` template) and passed as a configMap to the deployment.